link/kprobe.go

```diff
@@ -199,7 +199,7 @@ func kprobe(symbol string, prog *ebpf.Program, opts *KprobeOptions, ret bool) (*
 		return tp, nil
 	}
 	if err != nil && !errors.Is(err, ErrNotSupported) {
-		return nil, fmt.Errorf("creating perf_kprobe PMU: %w", err)
+		return nil, fmt.Errorf("creating perf_kprobe PMU (tried '%s'): %w", symbol, err)
```

Comment:
Maybe it's better to just return the original error?

```go
origErr := pmuKprobe(...)

fallbackErr := pmuKprobe(...)
if fallbackErr != nil { return origErr }
```

Reply:
The best would be both, being explicit about the fact that symbol1 and adaptedSymbol1 failed. Losing the fallbackErr could make the behavior not very clear I think.